pullman/storageproviders/gcs/provider.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2021 IBM Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package gcsprovider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/kserve/modelmesh-runtime-adapter/internal/util"
+	"github.com/kserve/modelmesh-runtime-adapter/pullman"
+)
+
+const (
+	configBucket      = "bucket"
+	configPrivateKey  = "private_key"
+	configClientEmail = "client_email"
+	configTokenUri    = "token_uri"
+
+	// only the "service_account" authentication type for GCS is supported
+	authTypeKey         = "type"
+	authTypeValue       = "service_account"
+	downloadConcurrency = 10
```

Comment:
Maybe rename to `maxDownloadConcurrency`?

I think it may be better for it to be smaller, e.g. 4, and to be able to override with an env variable.

Reply:
Sure, updated and reduced the default. Currently the s3 provider uses a download concurrency of 10, so this may need to be adjusted as well, along with making both providers support concurrency override with an env variable. I can open an issue for this.